aspnetcore/fundamentals/middleware/extensibility-third-party-container/sample/Middleware/SimpleInjectorMiddlewareFactory.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using Microsoft.AspNetCore.Http;
+using MiddlewareExtensibilitySample.Data;
+
+namespace MiddlewareExtensibilitySample.Middleware
+{
+    #region snippet1
+    public class SimpleInjectorMiddlewareFactory : IMiddlewareFactory
```

Comment:
This isn't right. The simple injector middleware should be using the simple injector container to resolve the IMiddleware instance. You shouldn't be newing it up.

Reply:
That's what I was asking on Slack: Merely registering the middleware in the SI container (even with the `SimpleInjectorMiddlewareFactory` in the SI container) doesn't activate it tho ... IIRC it throws an error when it finds a middleware that implements `IMiddleware` when the middleware isn't registered in the default container ... it clearly tries to use the default `MiddlewareFactory` to get it going. That's why you see me registering the `SimpleInjectorMiddlewareFactory` in the default container to get it going.

This *does* use the SI container for service resolution in the factory and the middleware. It's not clear how to do what ur asking to get the middleware running only using the SI container ... can you provide some code snippets in the sample, just paste some snippets in here, or provide more detailed instructions?

Reply:
ooooo ... wait a sec. I think I see what you mean. The middleware would be registered in the SI container ... resolved in the ctor here from the SI container ... THEN USED ... ok ok i see. 

Stand by ... I'll have a new commit up shortly.

Reply:
Not quite. Almost there 😉 . This factory is not for a single middleware, it is for *any* IMiddleware. That's why the type is passed into Create. That type needs to be passed into the simple injector container and resolve needs to be called to resolve the middlewareType from it.

Reply:
DAH! :smile: ... At least each step I get it a little closer. Ok ... that makes sense. I'll do that on the next commit.